tests/integration/test_guard_configurations.py

```diff
@@ -0,0 +1,319 @@
+        token=dr_token,
+        custom_model_id=custom_model_id,
+        guard_config_template_name=guard_template_name_toxicity,
+        guard_config_template_settings={
+            "deploymentId": guard_deployment_toxicity,
+        },
+        stages=["prompt"],
+        intervention=intervention,
+        name="pytest Toxicity Guard Configuration",
+    )
+
+    assert latest_version == latest_version_2
+
+    # ensure guard_template_name_toxicity appears only once
+    guard_config = _get_guard_configs(latest_version_2)
+    assert len([gc for gc in guard_config if guard_template_name_toxicity in gc["name"]]) == 1
+
+    latest_version_3 = get_update_or_create_guard_config_to_custom_model_version(
+        endpoint=dr_endpoint,
+        token=dr_token,
+        custom_model_id=custom_model_id,
+        guard_config_template_name=guard_template_name_prompt_injection,
+        guard_config_template_settings={
+            "deploymentId": guard_deployment_prompt_injection,
+        },
+        stages=["prompt"],
+        intervention=intervention,
+        name="pytest Prompt Injection Guard Configuration",
+    )
+
+    # Check the result
+    guard_config = _get_guard_configs(latest_version_3)
+    assert guard_config
+    assert any([guard_template_name_prompt_injection in gc["name"] for gc in guard_config]) and any(
+        [guard_template_name_toxicity in gc["name"] for gc in guard_config]
+    )
+
+    assert latest_version != latest_version_3
+
+    latest_version_4 = get_update_or_create_guard_config_to_custom_model_version(
+        endpoint=dr_endpoint,
+        token=dr_token,
+        custom_model_id=custom_model_id,
+        guard_config_template_name=guard_template_name_toxicity,
+        guard_config_template_settings={
+            "deploymentId": guard_deployment_toxicity,
+        },
+        stages=["prompt"],
+        intervention=intervention,
+        name="pytest Toxicity Guard Configuration",
+    )
+
+    assert latest_version_3 == latest_version_4
+    # ensure guard_template_name_toxicity appears only once
+    guard_config = _get_guard_configs(latest_version_4)
+    assert len([gc for gc in guard_config if guard_template_name_toxicity in gc["name"]]) == 1
+
+    latest_version_5 = get_update_or_create_guard_config_to_custom_model_version(
+        endpoint=dr_endpoint,
+        token=dr_token,
+        custom_model_id=custom_model_id,
+        guard_config_template_name=guard_template_name_toxicity,
+        guard_config_template_settings={
+            "deploymentId": guard_deployment_toxicity,
+        },
+        stages=["prompt"],
+        intervention=intervention_2,
+        name="pytest Toxicity Guard Configuration",
+    )
+
+    assert latest_version_4 != latest_version_5
+    # ensure guard_template_name_toxicity appears only once
+    guard_config = _get_guard_configs(latest_version_5)
+    assert len([gc for gc in guard_config if guard_template_name_toxicity in gc["name"]]) == 1
```

Comment:
think we should assert that the requested mutation to the intervention is also present server side here